ddpui/ddpdbt/dbt_service.py

```diff
@@ -2,48 +2,107 @@
 import shutil
 import subprocess
 from pathlib import Path
+import yaml
 
 import slugify
 
-from ddpui.ddpprefect import DBTCORE, SHELLOPERATION, prefect_service
-from ddpui.models.org import OrgDbt, OrgPrefectBlock, OrgWarehouse
+from ddpui.ddpprefect import (
+    DBTCORE,
+    SHELLOPERATION,
+    prefect_service,
+    DBTCLIPROFILE,
+    SECRET,
+)
+from ddpui.models.org import OrgDbt, OrgPrefectBlockv1, OrgWarehouse
 from ddpui.models.org_user import Org
-from ddpui.models.tasks import Task
+from ddpui.models.tasks import Task, OrgTask, DataflowOrgTask
+from ddpui.models.dbt_workflow import OrgDbtModel
 from ddpui.utils import secretsmanager
 from ddpui.utils.custom_logger import CustomLogger
+from dbt_automation import assets
 
 logger = CustomLogger("ddpui")
 
 
 def delete_dbt_workspace(org: Org):
     """deletes the dbt workspace on disk as well as in prefect"""
+
+    # we no longer have dbt core operation blocks
+    # for dbtblock in OrgPrefectBlock.objects.filter(org=org, block_type=DBTCORE):
+    #     try:
+    #         prefect_service.delete_dbt_core_block(dbtblock.block_id)
+    #     except Exception:  # pylint:disable=broad-exception-caught
+    #         pass
+    #     dbtblock.delete()
+
+    # we no longer have shell operation blocks
+    # for shellblock in OrgPrefectBlock.objects.filter(
+    #     org=org, block_type=SHELLOPERATION
+    # ):
+    #     if shellblock.block_name.find("-git-pull") > -1:
+    #         try:
+    #             prefect_service.delete_shell_block(shellblock.block_id)
+    #         except Exception:  # pylint:disable=broad-exception-caught
+    #             pass
+    #         shellblock.delete()
+
+    # remove transform tasks
+    org_tasks_delete = []
+    for org_task in OrgTask.objects.filter(org=org, connection_id__isnull=True).all():
+        if (
+            DataflowOrgTask.objects.filter(
+                orgtask=org_task, dataflow__dataflow_type="orchestrate"
+            ).count()
+            > 0
+        ):
+            raise Exception(f"{str(org_task)} is being used in a deployment")
+        org_tasks_delete.append(org_task)
+
+    logger.info("deleting orgtasks")
+    for org_task in org_tasks_delete:
+        for dataflow_orgtask in DataflowOrgTask.objects.filter(orgtask=org_task).all():
+            dataflow_orgtask.dataflow.delete()
+        org_task.delete()
```

Comment:
shouldn't we also delete the dataflow_orgtask

Reply:
that will happen automatically right ? datafloworgtask has `on_delete=models.CASCADE`